src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx

```diff
@@ -46,41 +45,19 @@ const CalculationConfig = (props) => {
   const { umap: umapSettings, tsne: tsneSettings } = data?.embeddingSettings.methodSettings || {};
   const { louvain: louvainSettings } = data?.clusteringSettings.methodSettings || {};
 
-  const debouncedClustering = useCallback(
-    _.debounce((resolution) => {
-      dispatch(runCellSetsClustering(experimentId, resolution));
-    }, 1500),
-    [],
-  );
-
-  const [resolution, setResolution] = useState(null);
   const [minDistance, setMinDistance] = useState(null);
 
-  useEffect(() => {
-    if (!resolution && louvainSettings) {
-      setResolution(louvainSettings.resolution);
-    }
-  }, [louvainSettings]);
-
   useEffect(() => {
     if (!minDistance && umapSettings) {
       setMinDistance(umapSettings.minimumDistance);
     }
   }, [umapSettings]);
 
-  const dispatchDebounce = useCallback(_.debounce((f) => {
-    dispatch(f);
-  }, 1500), []);
-
   const updateSettings = (diff) => {
-    if (diff.embeddingSettings) {
-      // If this is an embedding change, indicate to user that their changes are not
-      // applied until they hit Run.
-      onConfigChange();
-    } else {
-      // If it's a clustering change, debounce the save process at 1.5s.
-      dispatchDebounce(saveProcessingSettings(experimentId, FILTER_UUID));
-    }
+    // updates to configure embedding run on worker if they are the only changes
+    // need to know if change was to embedding or clustering settings
+    const settingType = Object.keys(diff)[0];
```

Comment:
Why only the first changed from the diff and not the whole `Object.keys(diff)`?

Reply:
Each of the diffs is a single object where the top level key is either going to be `embeddingSettings` or `clusteringSettings`. This just extracts that key.